#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.003006724593899898 -0.9982834959139409 -0.05848949814331818 3.0390466280309822
  position 0.35260907374229966 0.8909342754387352 -3.1839834404271703
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0.012062699038075102 0.07104350892963399 1.5664329347172739e-09
  rotation -1.9659416733418178e-07 1.876940876532297e-07 0.9999999999999629 -0.06408411542360215
  children [
    DEF GR_BODY Group {
      children [
        DEF ROBOT_BODY Shape {
          appearance PBRAppearance {
            baseColor 0.9294117647058824 0.8313725490196079 0
          }
          geometry DEF G_BODY Box {
            size 0.25 0.07 0.1
          }
        }
        Transform {
          translation 0.11 -0.05 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.12549019607843137 0.2901960784313726 0.5294117647058824
                roughness 1
                metalness 0
              }
              geometry Sphere {
                radius 0.015
              }
            }
          ]
        }
      ]
    }
    DEF WHEEL1_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position 0.9999999948323461
        axis 0 0 1
        anchor -0.05 0 0.06
      }
      device [
        PositionSensor {
          name "encoder1"
          resolution 1
        }
        RotationalMotor {
          name "motor_right"
          maxVelocity 8.33
          maxTorque 0.49
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation -0.05 0 0.06
        rotation 0.7913386724621613 0.43231028442149844 0.432308828789359 1.80272135790949
        children [
          DEF SH_WHEEL Shape {
            appearance DEF A_WHEEL PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry DEF G_WHEEL Cylinder {
              height 0.02
              radius 0.075
            }
          }
        ]
        boundingObject USE G_WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position -5.168227623641997e-09
        axis 0 0 1
        anchor -0.05 0 -0.06
      }
      device [
        RotationalMotor {
          name "motor_left"
          maxVelocity 8.33
          maxTorque 0.49
        }
      ]
      endPoint Solid {
        translation -0.05 0 -0.06
        rotation -1 2.5841138118209983e-09 -2.58410352886308e-09 1.5708003061004252
        children [
          Shape {
            appearance USE A_WHEEL
            geometry USE G_WHEEL
          }
        ]
        name "solid(1)"
        boundingObject USE G_WHEEL
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation 0.125 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.807843137254902 0.3607843137254902 0
          }
          geometry Box {
            size 0.02 0.01 0.01
          }
        }
      ]
      name "distance_sensor"
      lookupTable [
        0 0 0
        0.02 0 0
        11 1024 0
      ]
      resolution 1
    }
  ]
  boundingObject USE GR_BODY
  physics Physics {
  }
  controller "u2_wp"
}
Solid {
  translation 1.2199999221231974 0.19936615349481107 -7.787680319152741e-08
  rotation 0.7071067811914212 3.761541348248868e-11 -0.7071067811816738 -6.103621589479274e-07
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.12549019607843137 0.2901960784313726 0.5294117647058824
      }
      geometry DEF G_BOX Box {
        size 0.2 0.4 0.2
      }
    }
  ]
  boundingObject USE G_BOX
  physics Physics {
    density 101
  }
}
